### PR TITLE
Improve mux-simulator to support drop

### PR DESCRIPTION
**Pull request by wangxin**
### Description of PR

Summary:
Fixes # (issue)
This PR is to implement a new mux-simulator API for dropping packets to specified out ports. 

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
To fully support dualToR testing, we need to simulate packet drop condition in mux. This PR is to implement a new mux-simulator API for dropping packets to specified out ports. Tests can call this API to simulate packet dropping in mux.

#### How did you do it?
This change improved the mux-simulator to support a new API for dropping traffic to specified output ports. For example:
POST `/mux/<vm_set/<port_index>/drop` with data:
```
{"out_ports": ["nic", "tor_b"}
```
Then mux simulator will change the flow action to NIC port (PTF interface) and torB to "drop".

POST `/mux/<vm_set/<port_index>/output` with data:
```
{"out_ports": ["nic", "tor_a"}
```
Then mux simulator will change the flow action to NIC port (PTF interface) and torA to "output".

This change also enhanced the mux-simulator with logging capability for easier troubleshooting.

#### How did you verify/test it?
Call the API with different arguments by curl.

### CODE REVIEW

File: ansible/roles/vm_set/files/mux_simulator.py [hunk 1]
```diff
@@ -384,16 +403,169 @@ def all_mux_status(vm_set):
...
+        mux_status['bridge'],
+        in_port,
+        action_desc)
+    run_cmd(cmdline)
+    flow = {in_port: [{'action': action, 'out_port': out_port}]}
+    mux_status['flows'].update(flow)
+    return mux_status
+
+
+def update_flow_action_tor(mux_status, action, tor_ports):
```
Comment (bingwang-ms) [inline ansible/roles/vm_set/files/mux_simulator.py:455]: It's better to change the method name to ```update_flow_action_to_tor``` for better readability.

File: ansible/roles/vm_set/files/mux_simulator.py [hunk 2]
```diff
@@ -384,16 +403,169 @@ def all_mux_status(vm_set):
...
+    try:
+        assert 'out_ports' in data, 'Missing "out_ports" field'
+        for port in data['out_ports']:
+            assert port in supported_out_ports, 'Unsupported port: "{}", supported: {}'.format(port, supported_out_ports)
+        return True, ''
+    except Exception as e:
+        return False, 'Validate out_ports {} failed with exception: {}'.format(json.dumps(data), repr(e))
+
+
+def update_flow_action_nic(mux_status, action):
```
Comment (bingwang-ms) [inline ansible/roles/vm_set/files/mux_simulator.py:432]: It's better to change the method name to ```update_flow_action_to_nic``` for better readability.

Comment (wangxin) [reply ansible/roles/vm_set/files/mux_simulator.py]: Sure, updated.

File: ansible/roles/vm_set/files/mux_simulator.py (hunk 1 above)
Comment (wangxin) [reply ansible/roles/vm_set/files/mux_simulator.py]: Sure, updated.

**Review by theasianpianist — APPROVED** [review]

**Pull request merged**